src/Pmi/Datastore/Entity.php

```diff
@@ -1,229 +1,50 @@
-        $sql = 'SELECT * FROM ' . self::getSchema()->getKind();
-        $parameters = [];
-        if (count($criteria) > 0) {
-            $sql  .= ' WHERE ';
-            $conditions = [];
-            foreach ($criteria as $field => $value) {
-                $conditions[] = "{$field} = @{$field}";
-                $parameters[$field] = $value;
-            }
-            $sql .= implode(' AND ', $conditions);
-        }
-        return [$sql, $parameters];
+        $datastoreClient = new DatastoreClientHelper();
+        return $datastoreClient->upsert(static::getKind(), $id, $this->data);
     }
 
-    public static function fetchOneBy(array $criteria)
+    public static function delete($id)
     {
-        $datastore = new Datastore();
-        list($sql, $parameters) = static::buildSqlAndParameters($criteria);
-        $result = $datastore->fetchOneBySql(self::getSchema(), $sql, $parameters);
-        if ($result) {
-            $entity = new static();
-            $entity->gdsEntity = $result;
-            $entity->loadFromGdsEntity($result);
-            return $entity;
-        } else {
-            return null;
-        }
-    }
-
-    public static function fetchOneByKey($keyId) {
-        $datastore = new Datastore();
-        $result = $datastore->fetchOneByKey(self::getSchema(), $keyId);
-        if ($result) {
-            $entity = new static();
-            $entity->gdsEntity = $result;
-            $entity->loadFromGdsEntity($result);
-            return $entity;
-        } else {
-            return null;
-        }
-    }
-
-    public static function fetchOneByName($keyName) {
-        $datastore = new Datastore();
-        $result = $datastore->fetchOneByName(self::getSchema(), $keyName);
-        if ($result) {
-            $entity = new static();
-            $entity->gdsEntity = $result;
-            $entity->loadFromGdsEntity($result);
-            return $entity;
-        } else {
-            return null;
-        }
+        $datastoreClient = new DatastoreClientHelper();
+        return $datastoreClient->delete(static::getKind(), $id);
     }
 
-    public static function fetchBy(array $criteria)
-    {
-        $datastore = new Datastore();
-        list($sql, $parameters) = static::buildSqlAndParameters($criteria);
-        $results = $datastore->fetchAllBySql(self::getSchema(), $sql, $parameters);
-        $entities = [];
-        foreach ($results as $result) {
-            $entity = new static();
-            $entity->gdsEntity = $result;
-            $entity->loadFromGdsEntity($result);
-            $entities[] = $entity;
-        }
-        return $entities;
-    }
-    
-    public static function walkEntities(callable $callback, $sql, $params)
-    {
-        $datastore = new Datastore();
-        $datastore->walkEntities($callback, self::getSchema(), $sql, $params);
-    }
-    
-    public static function countEntities($sql, $params, $limit = 0)
+    public function gc($property, $value, $operator)
```

Comment:
This method for garbage collection is specific to the Session entity. I think it would make sense to break this up into two general-purpose methods.  One to query (with property, value, and operator) and one to delete a batch.